libs/langchain/langchain/storage/file_system.py

```diff
@@ -95,10 +101,15 @@ def mset(self, key_value_pairs: Sequence[Tuple[str, bytes]]) -> None:
         Returns:
             None
         """
-        for key, value in key_value_pairs:
-            full_path = self._get_full_path(key)
-            full_path.parent.mkdir(parents=True, exist_ok=True)
-            full_path.write_bytes(value)
+        orig_umask = os.umask(0)
```

Comment:
This sets the umask globally as far as I understand which will create a dangerous race condition with the rest of the code base and a security vulnurability for any file write anywhere.

File storage cache is meant mostly for prototyping purposes and should be getting getting extra functionality like this (even if it were done without a race condition)